git/show.go

```diff
@@ -0,0 +1,117 @@
+package git
+
+import (
+	"fmt"
+	"strings"
+)
+
+type FormatString struct {
```

Comment:
Okay, now that I understand what this is doing it explains some mysteries like why the pointer and how we could use flag for multiple options but this seems to be leaking implementation details of the flag package into library code. Can we make this just a type of string in the git package, but have a type in the cmd package to handle the flag interfaces? I think that would be cleaner.

Reply:
I'll give this some thought again tomorrow. I see where you're coming from.

Technically, there is no leaking of the flag package here into the git package. It just happens to conform to the interface defined there. Some of those functions may have limited usefulness outside of the context of flags. The choice of the pointer in the struct does seem to make it clear that there is an explicit unset state of nil, which I think is nice in some ways.

Maybe there's a cleaner way to separate the two concerns.

Reply:
Using pointers as Maybe types often leads to nil pointer dereferences down the line when someone forgets to check it, and always results in having two incompatible zero values (like on line 13).

I don't think this would be hard to fix. You just need to move your type def, Get, Set, and String methods to the cmd package as-is, then add a `type FormatString string` to the git package which keeps the remaining FormatCommit method  (but it would be simplified and safer by not having to check against nil or dereference the value.)